docs/DeveloperGuide.md

```diff
@@ -1,38 +1,506 @@
+      the `Order.status`.
+        * If the `Order.status` is already `refunded`, then the command would be invalid.
+          7.If the command passes all the validation checks, control is given back to `Refund` class and
+          the `Order.status` will be updated to `refunded` and is saved to the `orders.json` file using
+          the `Transaction.save` method. 
+6. Lastly, the control will be given back to the `Router` class and it then invokes the `Ui#printCommandSuccess` to
+   print a message indicating that the command has executed successfully.
+
+<hr>
+
+##### Basic Mode Refund an Order
 
-## Design & implementation
+This sequence diagram models the interaction between various components in MoneyGoWhere when the user inputs the
+command `refundorder` or `8`.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The general workflow of `refundorder` is as follows:
 
+1. User input is passed to `MoneyGoWhere`. `MoneyGoWhere` then passes it to the `Command` class, which instantiates a
+   new `parser` object to extract the command as `refundorder`.
+2. The 'Parser' object then uses `parser#formatInput` method from Parser class to extract all the arguments from the
+   user input.
+3. `Router#handleRoute` is then invoked to process the command. It calls the `Router#assistRoute` for the basic mode
+   commands.
+4. Once the command runs, it can be aborted at any time when the user inputs /cancel.
+4. The obtained command `refundorder` is then passed back to `MoneyGoWhere`, which instantiates a new `RefundAssistant`object
+   and calls the `RefundAssistant#refundOrder` method.
+5. `RefundAssistant#refundOrder` invokes the `getID` method to get and validate the order ID to be refunded (same validation as advanced mode).
+6.  If the command passes all the validation checks, control is given back to `Refund` class and the `Order.status` will be updated to `refunded` and is saved to the `orders.json` file using the `Transaction.save` method.
+7. Router#assistRoute then calls MenuAssistant#printResult to print a message indicating that if the item has been successfully added to the menu or if the user has cancelled the command accordingly.
 
-## Product scope
-### Target user profile
+<hr>
 
-{Describe the target user profile}
+### Report Commands
 
-### Value proposition
+#### Overview
 
-{Describe the value proposition: what problem does it solve?}
+* [Generate Sales Report](#add-an-item)
+* [Generate Ranking Report](#delete-an-item)
 
-## User Stories
+<hr>
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+## Requirements
 
-## Non-Functional Requirements
+### Project Scope
```

Comment:
Target user and value proposition could be included to understand the application idea better.